demos/TicTacToe/Classes/AppDelegate.cpp

```diff
@@ -22,7 +22,9 @@ bool AppDelegate::applicationDidFinishLaunching() {
 
   auto scene = TicTacToe::createScene();
   if (scene == NULL) {
-    exit(true);
+    free(director);
+    free(glview);
```

Comment:
free doesn't call C++ destructors.  You should call delete instead.

BUT... I'm not sure how you're supposed to release these things created by Cocoas2D.  It might not be enough to deallocate them - you might need to call a method on the class first. Like end() or destroy(), etc. You should check the exampes to see how they clean things up when the demo terminates.